pkg/controller/volume/expand/expand_controller.go

```diff
@@ -208,16 +208,21 @@ func (expc *expandController) pvcUpdate(oldObj, newObj interface{}) {
 		}
 
 		// Filter PVCs for which the corresponding volume plugins don't allow expansion.
-		volumeSpec := volume.NewSpecFromPersistentVolume(pv, false)
-		volumePlugin, err := expc.volumePluginMgr.FindExpandablePluginBySpec(volumeSpec)
-		if err != nil || volumePlugin == nil {
-			err = fmt.Errorf("didn't find a plugin capable of expanding the volume; " +
-				"waiting for an external controller to process this PVC")
-			expc.recorder.Event(newPVC, v1.EventTypeNormal, events.ExternalExpanding,
-				fmt.Sprintf("Ignoring the PVC: %v.", err))
-			glog.V(3).Infof("Ignoring the PVC %q (uid: %q) : %v.",
-				util.GetPersistentVolumeClaimQualifiedName(newPVC), newPVC.UID, err)
-			return
+		if volumeSpec := volume.NewSpecFromPersistentVolume(pv, false); volumeSpec.PersistentVolume.Spec.FlexVolume == nil {
+			// A flex volume is expandable but since it resides on the kubelet not the controller
```

Comment:
the volume driver shall be installed on master node.
According to https://github.com/kubernetes/community/blob/master/contributors/design-proposals/storage/grow-flexvolume-size.md#expandfs
```
If flex volume driver supports resizing, it shall implement expandvolume method and at least, the volume driver shall be installed on master node.
```

Reply:
Did we finalize on that? I don't particularly like special casing here for flex. Can we not make it a hard requirement that for flex volume resizing the plugin must be at least installed on the controller so as controller is aware of plugin capabilities before allowing volume expansion?

Reply:
@gnufied I was under the impression there was discussion around not make that (installing flex plugin on master) a hard requirement. Maybe we need to revisit it? 

If we do make it a hard requirement, it would lead to issues for certain uses cases. Esp. for "managed Kubernetes service" like [Amazon EKS](https://aws.amazon.com/eks/) or [IBM CKS](https://www.ibm.com/cloud/container-service) or [AKS](https://azure.microsoft.com/en-us/services/kubernetes-service) - in each cases the control management is hidden from the end user making it tough to install the flex driver on the master.

How about an alternative -

1. The volume-expand controller sets the PVC condition to `ExternalResizing` instead of `Resizing` for plugins (only Flex today) that don't reside on the master/controller
2. The kubelet reconciler calls `ExpandVolumeDevice` of the plugin (Flex) and then 
  - if it succeeds the condition is cleared prompting the volume-expand controller to finish and update PV and move onto fs resizing as it does today
  - if it fails the condition is updated to `ExternalResizingFailed` which prompts the volume-expand to abort resizing with appropriate error

This kind of controller <-> kubelet feedback loop is necessary since only the controller is guaranteed to have privileges to update PV, the kubelet may not be able to do it.

Reply:
@saad-ali @chakri-nelluri what you guys think about this? I think that - controller to kubelet and then back to controller feedback loop is problematic. Expand Controller must reprocess such PVCs.

Reply:
@saad-ali @chakri-nelluri - @gnufied and I were discussing this on slack, and there maybe a simpler way.

This discussion got underway because the design mentioned the need to install Flex on the controller so that a Flex author can signal expansion is unsupported by returning an error from `ExpandVolumeDevice`. 

Since only dynamically provisioned volumes with `allowVolumeExpansion = true` in their storage classes are allowed to be expanded, a Flex plugin that doesn't support expansion can simply set `allowVolumeExpansion = false` and we don't need to come up with a complicated flow to handle it.

So is that a good enough option?

Reply:
@aniket-s-kulkarni if the plugin is not installed on master, assume it to be true and mark the volume "RequiresFSResize" and let the Kubelet side code manage it. Let me know  if you see any issues with it?

Reply:
@aniket-s-kulkarni The code freeze is very close. Ping me on slack if you need a quick review.

Reply:
@chakri-nelluri you are right. That works - and that's how the PR is doing it. And with what @gnufied mentioned above - it covers all the cases. I will address your comment about abstracting the `.FlexVolume != nil` checks